### PR TITLE
docs: richer documentation

### PR DESCRIPTION
**Pull request by tiborsimko**
* Amends Installation part to separate the installation of client, server, and
  supporting cloud infrastructure.

* Adds detailed "Getting Started" guide with walk-trough instructions from how
  to set up REANA on a laptop, through running simple demo examples, to how to
  clean after ourselves.

* Adds the new component reana-message-broker to the list of components.

Signed-off-by: Tibor Simko <tibor.simko@cern.ch>

### CODE REVIEW

**Comment by tiborsimko** [comment]
@diegodelemos @hjhsalo This is for our "live" naming discussion... still needs work.

File: docs/gettingstarted.rst [hunk 1]
```diff
@@ -1,16 +1,149 @@
...
+<https://kubernetes.io/docs/getting-started-guides/minikube/>`_. How to do this
+depends on your operating system. For example, on Arch Linux, you can install
+the following packages:
+
+- `virtualbox <https://www.archlinux.org/packages/community/x86_64/virtualbox/>`_
+- `virtualbox-guest-iso <https://www.archlinux.org/packages/community/x86_64/virtualbox-guest-iso/>`_
+- `virtualbox-host-modules-arch <https://www.archlinux.org/packages/community/x86_64/virtualbox-host-modules-arch/>`_
+- `minikube <https://aur.archlinux.org/packages/minikube/>`_ from AUR
+- `kubectl-bin <https://aur.archlinux.org/packages/kubectl-bin/>`_ from AUR
+
```
Comment (diegodelemos) [inline docs/gettingstarted.rst]: We should add `docker` client as a requirement (open issue https://github.com/reanahub/reana/issues/18)

File: docs/gettingstarted.rst [hunk 2]
```diff
@@ -1,16 +1,149 @@
...
+Run "word population" example analysis
+--------------------------------------
+
+You can now submit a `"world population" example analysis
+<https://github.com/reanahub/reana-demo-worldpopulation>`_ to run on your local
+REANA cloud:
+
+.. code-block:: console
+
+   $ reana run reanahub/reana-demo-worldpopulation -a atlas
```
Comment (diegodelemos) [inline docs/gettingstarted.rst]: `-e` for experiment

Comment (tiborsimko) [inline docs/gettingstarted.rst]: Yup, typo.  But actually I'd rather remove it and make it optional. First, "experiment" is CERN-centric. Second, we don't support it well today anyway. So I'd amend `reana` script and make it optional as part of this PR.

File: docs/gettingstarted.rst (hunk 1 above)
Comment (tiborsimko) [inline docs/gettingstarted.rst]: BTW no need to open new issue, I'll fix it here.

**Pull request merged**